op-e2e/sequencer_failover_setup.go

```diff
@@ -299,24 +299,56 @@ func sequencerCfg(rpcPort int) *rollupNode.Config {
 	}
 }
 
-func waitForLeadershipChange(t *testing.T, c *conductor, leader bool) error {
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+func waitForLeadership(t *testing.T, c *conductor) error {
+	condition := func() (bool, error) {
+		isLeader, err := c.client.Leader(context.Background())
+		if err != nil {
+			return false, err
+		}
+		return isLeader, nil
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 	defer cancel()
-	for {
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		default:
-			isLeader, err := c.client.Leader(ctx)
-			if err != nil {
-				return err
-			}
-			if isLeader == leader {
-				return nil
-			}
-			time.Sleep(500 * time.Millisecond)
+	return wait.For(ctx, 10*time.Second, condition)
+}
+
+func waitForLeadershipChange(t *testing.T, prev *conductor, prevID string, conductors map[string]*conductor, sys *System) string {
+	condition := func() (bool, error) {
+		isLeader, err := prev.client.Leader(context.Background())
+		if err != nil {
+			return false, err
 		}
+		return !isLeader, nil
 	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	err := wait.For(ctx, 10*time.Second, condition)
```

Comment:
should we reduce the check interval from 10 to like 1?

Reply:
Good call; lowered these